sdk/storage/test/test_base.hpp

```diff
@@ -29,9 +29,14 @@ namespace Azure { namespace Storage { namespace Test {
 
   void RandomBuffer(char* buffer, std::size_t length);
 
-  std::vector<uint8_t> ReadBodyStream(std::unique_ptr<Azure::Core::Http::BodyStream>& stream);
+  inline std::vector<uint8_t> ReadBodyStream(std::unique_ptr<Azure::Core::Http::BodyStream>& stream)
+  {
+    Azure::Core::Context context;
+    return *Azure::Core::Http::BodyStream::ReadToEnd(context, *stream);
+  }
 
-  inline std::vector<uint8_t> ReadBodyStream(std::unique_ptr<Azure::Core::Http::BodyStream>&& stream)
+  inline std::vector<uint8_t> ReadBodyStream(
+      std::unique_ptr<Azure::Core::Http::BodyStream>&& stream)
```

Comment:
We would not need these ones anymore.  Update to use the BodyStream::ReadToCount(...) or BodyStream::ReadToEnd(..) functions

Reply:
I think so

Reply:
> @JinmingHu-MSFT , yes, 9 and 10 are fixed.
> 
> And the seg fault was an error in the sample test cases. I have fixed.
> 
> I was getting the hang while reading and found the root cause. I have tested Core transport implementation with the samples I have and it is working fine with the changes on this PR.
> 
> Please fix Storage Code whenever possible
> 
> About the BodyStream Reference. Core team will come to a final decision about this tomorrow. (I will update PR after that)

9 isn't fixed.

Here https://github.com/Azure/azure-sdk-for-cpp/blob/74c84147c5b8d673ecddb58926704ae92ed49ea5/sdk/core/azure-core/src/http/curl/curl.cpp#L444,

if I remove Line 444, then 9 is fixed.

I didn't look into it carefully, so please double check the logic here is correct.

Reply:
Now it's fixed